bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -184,136 +225,432 @@ static void deleteBcStateTransferDbFolder(const string& path) {
   }
 }
 
+static DataStore* createDataStore(const string& dbName, const Config& targetConfig) {
```

Comment:
Should we have smart pointer here and also other places?

Reply:
We can't have a smart pointer at this stage since this object is managed raw inside ST code